cpp/splinepy/splines/helpers/properties.hpp

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <array>
+
+namespace splinepy::splines::helpers {
+
+/// Returns parametric bounds as (para_dim x 2) array.
+template<typename SplineType>
+inline std::array<std::array<double, SplineType::kParaDim>, 2>
+GetParametricBounds(const SplineType& spline) {
+  std::array<std::array<double, SplineType::kParaDim>, 2> parametric_bounds{};
+
+  if constexpr (!SplineType::kHasKnotVectors) {
+    parametric_bounds[0].fill(0.);
+    parametric_bounds[1].fill(1.);
+  } else {
+    const auto& parameter_space = spline.GetParameterSpace();
+    size_t i{};
+    for (const auto& knotvector : parameter_space.GetKnotVectors()) {
+      parametric_bounds[0][i] = knotvector->GetFront().Get();
+      parametric_bounds[1][i] = knotvector->GetBack().Get();
+      ++i;
+    }
+  }
+
+  return parametric_bounds;
+}
+
+template<typename SplineType>
+inline int GetNumberOfSupports(const SplineType& spline) {
```

Comment:
I personally would prefer an unsigned return type here, but that obviously depends on where its needed.